examples/kotlin/benchmarks/src/TestInheritedBenchmark.kt

```diff
@@ -0,0 +1,34 @@
+package test
+
+import org.openjdk.jmh.annotations.*
+import java.util.concurrent.TimeUnit
+import kotlin.math.cos
+import kotlin.math.sqrt
+
+@State(Scope.Benchmark)
+@Fork(1)
+@Warmup(iterations = 0)
+@Measurement(iterations = 1, time = 1, timeUnit = TimeUnit.SECONDS)
+abstract class BaseBenchmark {
```

Comment:
Can the base class declare benchmark methods as well?

Reply:
No. As to why? -> It doesn't make sense to run the same benchmark again if two or more classes are inheriting from same non-final class.

Reply:
I can think of the following use case:
```
import kotlinx.benchmark.*

@State(Scope.Benchmark)
@Measurement(iterations = 7, time = 1, timeUnit = BenchmarkTimeUnit.SECONDS)
@Warmup(iterations = 7, time = 1, timeUnit = BenchmarkTimeUnit.SECONDS)
@OutputTimeUnit(BenchmarkTimeUnit.MILLISECONDS)
abstract class IterationBenchmarkBase {
    abstract val list: List<String>

    @Param("10", "100")
    var size: Int = 0

    @Benchmark
    fun iterationBenchmark(bh: Blackhole) {
        for (element in list) {
            bh.consume(element)
        }
    }
}

class ArrayListIterationBenchmark : IterationBenchmarkBase() {
    override val list: List<String>
        get() = ArrayList<String>(size).apply { fill(size) }
}

class ArrayDequeIterationBenchmark : IterationBenchmarkBase() {
    override val list: List<String>
        get() = ArrayDeque<String>(size).apply { fill(size) }
}

private fun MutableList<String>.fill(size: Int) {
    addAll((1..size).map { it.toString() })
}
```